### PR TITLE
Make it possible to enable encryption at host for a VM and add test cases

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#38422](https://github.com/Azure/azure-sdk-for-java/issues/38422) and [pr#38409](https://github.com/Azure/azure-sdk-for-java/pull/38409)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:2542]: stage of the VM update ?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:2546]: Just Enables/Disabled in the Javadoc. No need for the "Request to".

Same to the create flow (if the contributor uses "Request to").

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java]: Fixed in the new version.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java]: Enables , plural

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java]: I mean the Javadoc

**Review by weidongxu-microsoft — COMMENTED** [review]
Personally, I'd go with `withEncryptionAtHost` etc., as we already approved that via contributor's PR (having no opinion on either side).

@XiaofeiCao for input as well.

**Comment by XiaofeiCao** [comment]
> Personally, I'd go with `withEncryptionAtHost` etc., as we already approved that via contributor's PR (having no opinion on either side).
> 
> @XiaofeiCao for input as well.

+1 for `withEncryptionAtHost` to align with the contributor. 
On the other hand, if we want to go with `enableEncryptionAtHost`, we better rename `isEncryptionAtHost` to `isEncryptionAtHostEnabled` as well.

**Review by weidongxu-microsoft — APPROVED** [review]
Please wait for another approval

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**
